spec/query/query_string_spec.rb

```diff
@@ -0,0 +1,155 @@
+    let(:condition) { "filter text" }
+    let(:expected_term) { '(*filter AND text*)' }
+
+    shared_examples "nothing" do
+      it "does not include q" do
+        expect(term).to be_nil
+      end
+    end
+
+    shared_examples "a search term" do
+      it "is converted to an elasticsearch term" do
+        expect(term).to eq(expected_term)
+      end
+    end
+
+    context "when condition is a String" do
+      it_behaves_like "a search term"
+
+      context "condition is empty string" do
+        let(:condition) { "" }
+        it_behaves_like "nothing"
+      end
+
+      context "with extra spaces" do
+        let(:condition) { "filter    text" }
+        it_behaves_like "a search term"
+      end
+
+      context "escaping special characters" do
+        let(:condition) { '~tex*t wi[t]h "s{pec}ial" #v@lue$!^' }
+        let(:expected_term) { '(*\~tex\*t wi\[t\]h \"s\{pec\}ial\" #v@lue$s!*)\^' }
+      end
+    end
+
+    context "when condition is a Hash" do
+      let(:condition) { { some: { nested: { condition: "filter text" } } } }
+      let(:expected_term) { 'some.nested.condition:(*filter AND text*)' }
+
+      it_behaves_like "a search term"
+
+      context "with many terms" do
+        let(:condition) { { some: { nested: { condition: "filter text", part: "two" } } } }
+        let(:expected_term) { 'some.nested.condition:(*filter AND text*) AND some.nested.part:(*two*)' }
+
+        it_behaves_like "a search term"
+      end
+
+      context "value is an Array" do
+        let(:condition) { { some: { nested: { condition: ["filter", "text"] } } } }
+        let(:expected_term) { 'some.nested.condition:((*filter*) OR (*text*))' }
+
+        it_behaves_like "a search term"
+      end
+
+      context "value is a Range" do
+        let(:condition) { { some: { nested: { condition: (1..2) } } } }
+        let(:expected_term) { "some.nested.condition:[1 TO 2]" }
+
+        it_behaves_like "a search term"
+      end
+    end
+
+    context "when condition is an Array" do
+      let(:condition) { ["filter text", "some more filter text"] }
+      let(:expected_term) { '((*filter AND text*) OR (*some AND more AND filter AND text*))' }
+
+      it_behaves_like "a search term"
+    end
+
+    context "when condition is a Numeric" do
+      let(:condition) { 1 }
+      let(:expected_term) { 1 }
+
+      it_behaves_like "a search term"
+    end
+
+    context "when condition is a Range" do
+      context "that is inclusive" do
+        let(:condition) { 1..3 }
+        let(:expected_term) { "[1 TO 3]" }
+
+        it_behaves_like "a search term"
+      end
+
+      context "that is exclusive" do
+        let(:condition) { 1...3 }
+        let(:expected_term) { "[1 TO 2]" }
+
+        it_behaves_like "a search term"
+      end
+
+      context "values are integers" do
+        it_behaves_like "a search term"
+      end
+
+      context "values are strings" do
+        let(:condition) { 'a'..'z' }
+        let(:expected_term) { "[a TO z]" }
```

Comment:
Should there be a test for an exclusive string range?

Reply:
see me previous comment